include/dd/Package.hpp

```diff
@@ -1662,6 +1658,22 @@ namespace dd {
                             return e;
                         }
                     }
+                } else {
+                    // If level is lower, then that means there is
+                    // an identity at the current level of that DD
+                    if (x.p->v > y.p->v) {
+                        e = xCopy;
+                    } else {
+                        e = yCopy;
+                    }
+                    computeTable.insert(xCopy, yCopy, {e.p, e.w});
+                    e.w = cn.mulCached(x.w, y.w);
+
+                    if (e.w.approximatelyZero()) {
+                        cn.returnToCache(e.w);
+                        return ResultEdge::zero;
+                    }
+                    return e;
```

Comment:
this seems wrong to me. Just because there is an identity at the current level does not mean that the result of the multiplication is just the other DD. This should still require recursive calls to properly handle the multiplication.